SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -469,41 +536,48 @@ void FSpatialGDKEditorToolbarModule::LaunchInspectorWebpageButtonClicked()
 	}
 }
 
-bool FSpatialGDKEditorToolbarModule::StartSpatialOSStackCanExecute() const
+bool FSpatialGDKEditorToolbarModule::StartSpatialDeploymentIsVisible()
 {
-	return !SpatialOSStackProcHandle.IsValid() && !FPlatformProcess::IsApplicationRunning(SpatialOSStackProcessID);
+	return (LocalDeploymentManager->IsSpatialServiceRunning() && !LocalDeploymentManager->IsLocalDeploymentRunning()) || !LocalDeploymentManager->IsSpatialServiceRunning();
```

Comment:
I'm finding it hard to read this. Might be easier to parse as some nested ifs?
```
if (LocalDeploymentManager->IsSpatialServiceRunning())
{
  return !LocalDeploymentManager->IsLocalDeploymentRunning();
}
else
{
  return true;
}
```
Happy to be overruled though.

Reply:
Yeah sure I don't mind.